### DISCUSSION TITLE
`coerce_path()` in `SchemaGenerator` will not get correct pk field if `get_queryset()` method is used on view

### DISCUSSION THREAD

**Discussion by mhamid3d** [encode/django-rest-framework#8325] [Potential Issue]
The `coerce_path` method is trying to grab the `queryset` attribute from the view. If you are using `get_queryset` on the view, it will not satisfy the `if model` and then default to `id`.

```python
def coerce_path(self, path, method, view):
    """
    Coerce {pk} path arguments into the name of the model field,
    where possible. This is cleaner for an external representation.
    (Ie. "this is an identifier", not "this is a database primary key")
    """
    if not self.coerce_path_pk or '{pk}' not in path:
        return path
    model = getattr(getattr(view, 'queryset', None), 'model', None)
    if model:
        field_name = get_pk_name(model)
    else:
        field_name = 'id'
    return path.replace('{pk}', '{%s}' % field_name)
```

**Comment by lovelydinosaur** [comment]
Can you provide a simple-as-possible example that demonstrates the issue?

**Reply by mhamid3d** [reply]
For example, if I was using `uuid` instead of `id` as my primary key, and using `get_queryset` on my view.

`app/models.py`
```python
class Product(models.Model):
    uuid = models.UUIDField(primary_key=True)
```

`app/views.py`
```python
class ProductView(viewsets.ModelViewSet):

    def get_queryset(self):
        qs = models.Product.objects.all()
        # do stuff
        return qs
```
Then generate a schema:

```python
from rest_framework.schemas.openapi import SchemaGenerator

print(SchemaGenerator().get_schema())
```

Resulting schema will have paths such as:

`app/product/{id}`

instead of

`app/product/{uuid}`

If it doesn't make a difference to add a `queryset = ...` in addition to the `get_queryset` method, then maybe it should be suggested or disclaimer in the schema docs.

**Reply by lovelydinosaur** [reply]
Seems reasonable to me.

If you'd like to try a pull request for this, then we can see if it introduces any unexpected effects vs. the test suite. If not then I think this would go in, yes.
